src/services/mapService.js

```diff
@@ -35,5 +36,50 @@ export default {
     return axios.get(`/paths/${pathId}/stations`).catch(() => {
       store.dispatch("showError");
     });
+  },
+
+  getPathToDraw(pathId) {
+    return axios.get(`paths/${pathId}/stations`).then(response => {
+      const stations = response.data;
+      const endStation = stations.pop().position;
+      const startStation = stations.shift().position;
+      stations.forEach(
+        (station, index) =>
+          (stations[index] = { location: station.position, stopover: true })
+      );
+      return {
+        startStation: startStation,
+        endStation: endStation,
+        waypoints: stations
+      };
+    });
+  },
+
+  drawRoute(
+    directionsService,
+    directionsDisplay,
+    startStation,
+    endStation,
+    waypoints
+  ) {
+    directionsService.route(
+      {
+        origin: startStation,
+        destination: endStation,
+        waypoints: waypoints,
+        travelMode: "WALKING"
+      },
+      function(response, status) {
+        if (status === "OK") {
+          directionsDisplay.setOptions({ suppressMarkers: true });
+          directionsDisplay.setDirections(response);
+        } else {
+          window.alert(
+            "Directions request failed, try again later. " +
+              "If the problem repeats feel free contact with us"
+          );
+        }
+      }
```

Comment:
When only you can, use arrow function instead of older syntax function(){}. An arrow function is more concise and easier to read.